frontend/src/components/Playback/Autoplay.js

```diff
@@ -1,108 +1,78 @@
 import React, { useRef, useState, useEffect } from "react";
 
 import * as audio from "../../util/audio";
-import { MEDIA_ROOT } from "../../config";
+import * as webAudio from "../../util/webAudio";
 
 import Circle from "../Circle/Circle";
 import ListenCircle from "../ListenCircle/ListenCircle";
-import Preload from "../Preload/Preload";
-
-const PRELOAD = "PRELOAD";
-const RECOGNIZE = "RECOGNIZE";
-
 
 const AutoPlay = ({instruction, preloadMessage, onPreloadReady, playConfig, sections, time, startedPlaying, finishedPlaying, responseTime, className=''}) => {
     // player state
-    const [state, setState] = useState({ view: PRELOAD });
+    
     const running = useRef(playConfig.auto_play);
-    const setView = (view, data = {}) => {
-        setState({ view, ...data });
-    }
-
+    
     const section = sections[0];
 
     const onCircleTimerTick = (t) => {
         time.current = t;
     };
 
-
     // Handle view logic
-    useEffect(() => {
-        switch (state.view) {
-            case RECOGNIZE:                
-                // Play audio at start time            
-                if (!playConfig.mute) {
-                    audio.playFrom(Math.max(0, playConfig.playhead));
-                }                
-                startedPlaying();                
-                break;
-            default:
-            // nothing
+    useEffect(() => {        
+        let latency = 0;
+        // Play audio at start time            
+        if (!playConfig.mute) {
+            if (playConfig.play_method === 'BUFFER' && !playConfig.external_audio) {
+                console.log('Autoplay buffer');
+                latency = webAudio.getTotalLatency();
+                webAudio.playBufferFrom(section.id, 0, Math.max(0, playConfig.playhead));
```

Comment:
Won't this always play the file from start? (Second argument is 0)

Reply:
That's indeed a typo.